workspaces/ui-v2/src/optic-components/hooks/useEndpointBodyHook.ts

```diff
@@ -47,27 +51,57 @@ export function useEndpointBody(
         contentType
         rootShapeId
       }
-      responses {
-        id
-        statusCode
-        contributions
-        changes(sinceBatchCommitId: $sinceBatchCommitId) {
-          added
-          changed
-        }
-        bodies {
-          contentType
-          rootShapeId
-        }
-      }
     }
-    }`,
+  }
+}`;
+
+type Body = {
+  contentType: string;
+  rootShapeId: string;
+};
+
+type EndpointBodyQueryResponse = {
+  requests: {
+    id: string;
+    pathId: string;
+    method: string;
+    requestContributions: Record<string, string>;
+    bodies: Body[];
+    changes?: IChanges;
+    responses: {
+      id: string;
+      statusCode: number;
+      contributions: Record<string, string>;
+      changes?: IChanges;
+      bodies: Body[];
+    }[];
+  }[];
+};
+
+export function useEndpointBody(
+  pathId: string,
+  method: string,
+  renderChangesSince?: string
+): { requests: IRequestBody[]; responses: IResponseBody[] } {
```

Comment:
There's probably some "cool" things we could do here with generics where if we specify `renderChangesSince` as a string, it'll return us changelog information - but I think splitting out the doc page from the contributions page is probably a better idea (since the doc page could diverge in functionality, especially with manual edits) meaning that we wouldn't need this to be generic